interoperability-tests/Testing-between-Gramine-and-IntelSGXSDK-ratls.md

```diff
@@ -0,0 +1,158 @@
+Testing between Gramine and rats-tls
+====
+
+# Introduction
+
+This document records the interoperability testing between [Gramine](https://github.com/gramineproject/gramine) and [Intel SGXSDK](https://github.com/intel/linux-sgx), both of which have implemented Interoperable RA-TLS.
+
+We conducted two sets of tests, in which Gramine and Intel SGXSDK ra-tls respectively verified each other's certificates and established TLS secure channels. You can verify our tests by following the steps below.
+
+# Prerequisites
+
+## Intel SGXSDK 
+
+To run the tests, first you need to download and install Intel SGXSDK and corresponding PSW packages on your system. 
+
+Interoperable RA-TLS was implemented since Intel SGXSDK 2.22 release. 
+Download the software package from following link:
+https://download.01.org/intel-sgx/latest/linux-latest/distro/
+
+Pick up your the distro for your OS. 
+
+Install Intel SGXSDK: sgx_linux_x64_sdk_2.22.100.3.bin
+
+Install corresponding PSW packages: sgx_debian_local_repo.tgz
+
+# Intel SGXSDK ra-tls as attester and Gramine as verifier
+
+## Intel SGXSDK ra-tls
+
+Intel SGXSDK ra-tls provides a sample program [SampleAttestedTLS](https://github.com/intel/linux-sgx/tree/master/SampleCode/SampleAttestedTLS), which can be used to start an RA-TLS server.
+
+
+1. Get source code of SampleAttestedTLS
+
+```sh
+git clone https://github.com/intel/linux-sgx
+```
+
+2. Build the sample
+
+```sh
+cd SampleCode/SampleAttestedTLS
+
+make
+```
+3. Start the server
+
+```sh
+./server/host/tls_server_host ./server/enc/tls_server_enclave.signed.so -port 4433
+
+```
+
+Now the rats-tls-server listens on `127.0.0.1:4433`, with a self-signed X.509 certificate.
+
+Here is a certificate generated by Intel SGXSDK ra-tls-server: [intel-sgxsdk-ra-tls-cert.pem](./intel-sgxsdk-ra-tls.pem)
```

Comment:
What is the point of providing this certificate? The cert is anyway self-signed and automatically generated, so it will be different every time.

Is it just as an example, so that people can examine how the certificates look like? I like this idea, but in this case, please reword your text to: `Here is an example certificate generated ...`

Reply:
updated